portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/* nav buttons */
+
+var sections = ['.home', '.projects', '.cats', '.skills', '.contacts']
+
 /**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+ * Return function, which moves to the section with index 'sectionIndex' and,
+ * if the navbar ('nav') will hide part of the
+ * section's contect, scrolls up by the navbar height.
+ * @param  {int} sectionIndex [index number of the section]
+ * @return {function}         [callback function for the event]
+*/
+function makeScrollCallBack(sectionIndex) {
+    return function() {
+        var sectionDOM = document.querySelector(sectionIndex)
+        sectionDOM.scrollIntoView(); 
+        
+        var height = document.querySelector('nav').clientHeight;
+        var scrolledY = window.scrollY;
+        if (scrolledY === sectionDOM.offsetTop) {
+            window.scroll(0, scrolledY - height);
+        }
+    }
+}
+
+document.querySelector('.nav-logo').addEventListener('click', makeScrollCallBack('.home-section'));
+
+for (var i = 0; i < sections.length; i++) {
+    document.querySelector('.nav-item' + sections[i]).addEventListener('click', makeScrollCallBack(sections[i] + '-section'));
 }
+
+/* projects buttons */
+
+var listOfProjects = document.querySelector('.projects-names').childElementCount;
+
+/**
+ * Return function, which hides old description (which has class 'active')
+ * and replaces it with the new chosen,
+ * changes styles for chosen '.project-name' and old '.project-name.active'.
+ * @param  {int} projectIndex [index number of the project name and its description, respectively]
+ * @return {function}         [callback function for the event]
+*/
+function makeChangeDescriptionCallBack(projectIndex) {
+    return function() {
+        document.querySelector('.project-name.active').classList.remove('active');
+        document.querySelector('.projects-names').children[projectIndex].classList.add('active');
+        
+        document.querySelector('.project-description.active').classList.remove('active');
+        document.querySelector('.projects-descriptions').children[projectIndex].classList.add('active');
+    }
+}
```

Comment:
It is a good practice to declare all functions together, so I would move this to the top of the file, close to makeScrollCallBack.

Then the rest of your code would also be together and easier to read, since there will be no functions declaration between statements.